Document TrustSDK specs (#121)
trustwallet/developer: trust-sdk/trust-sdk-specs.md

## URL Specifications

> TrustSDK works by calling deeplinks between apps, this document describes URL formats for each command. Due to the SDK high customization provided by TrustSDK, clients and wallets can adapt the URLs to their own needs. 

Reviewer: ```suggestion
TrustSDK works by calling deeplinks between apps, this document describes URL formats for each command. TrustSDK is highly customizable, clients and wallets can adapt the URLs to their own needs. 
```

> TrustSDK works by calling deeplinks between apps, this document describes URL formats for each command. Due to the SDK high customization provided by TrustSDK, clients and wallets can adapt adapt the URL to it's own needs. 

Reviewer 2: > clients and wallets can adapt adapt the URL to it's own needs

clients and wallets can adapt the URLs to their own needs

All commands follow the URL format below:

```
[scheme]://[command]?[params]&[app]&[id]&[callback]
```

Where: 
* `[scheme]` is the wallet's deeplink scheme. Default to `trust`
* `[command]` is the command to be executed
* `[params]` is the command parameters encoded as url query items.
* `[id]` is the unique command id. The value is later used to resolve the callback for the command. Can be any incrementing integer.
* `[app]` callback URL scheme. 
* `[callback]`callback path. Default to `sgn_sign_result`

### Sign

> Sign command is used to sign and broadcast transactions following [wallet-core](/wallet-core/wallet-core.md) protobuf models encoded in base64.

Reviewer: ```suggestion
Sign command is used to sign and broadcast transactions leveraging [wallet core](/wallet-core/wallet-core.md) protobuf models (encoded in base64).
```

#### Command
* Command: `sdk_sign`
* Parameters:
  * `coin`: BIP44 coin code

>   * `data`: wallet-core protobuf model binary encoded in base64

Reviewer: ```suggestion
  * `data`: wallet core protobuf model binary encoded in base64
```

>   * `send`: flag that indicates if the wallet should broadcast the transaction. Possible values: `true`, `false`.

Reviewer: ```suggestion
  * `send`: flag that indicates if the wallet should broadcast the transaction. Valid values: `true`, `false`.
```

  * `meta`: transaction metadata. This attribute is optional and encoded as a dictionary. Check how it's encoded in [Dictionary Encoding](#dictionary-encoding).

>     * `meta.__name`: transaction metadata type. Possible values: `dApp`

Reviewer: dApp or dapp?
```suggestion
    * `meta.__name`: transaction metadata type. Possible values: `dapp`?
```

Author: `dApp`

Reviewer: the example request is still `dapp`

Reviewer 2: case-insensitive check is a good idea in such cases (just to consider)

    * `meta.name`: dapp name
    * `meta.url`: dap url encoded

##### Example

```
trust://sdk_sign?coin=60&data=ChQAAAAAAAAAAAAAAAAAAAAAAAAAARIUAAAAAAAAAAAAAAAAAAAAAAAAA
d0aFAAAAAAAAAAAAAAAAAAAAAB94pAAIhQAAAAAAAAAAAAAAAAAAAAAAABSCCoqMHg3MjhCMDIzNzcyMzBiNWRm
NzNBYTRFMzE5MkU4OWI2MDkwREQ3MzEyMhQAAAAAAAAAAAAAAAAAAFrzEHpAAA&meta.__name=dapp
&meta.name=Test&meta.url=https://dapptest.com&send=false
&app=sampleapp&callback=sdk_sign_result&id=1
```

#### Response
* Parameters:
  * `coin`: BIP44 coin code
  * `data`: transaction signed as base64 or the transaction hash if `send` is `true`

##### Example

```
sampleapp://sdk_sign_result?coin=60&data=Cm34a4IB3YR94pAAglIIlHKLAjdyMLXfc6pOMZLom2CQ3XMSh
lrzEHpAAIAmoB0Yegc2ZqxtkSkXlYo_TEg4eBrjopGUj9ySxJh6JlfToGqR7yNKzV8cD_yN_jVR5YrVaTANO05X2_
9HleO8htQqEgEmGiAdGHoHNmasbZEpF5WKP0xIOHga46KRlI_cksSYeiZX0yIgapHvI0rNXxwP_I3-NVHlitVpMA0
7Tlfb_0eV47yG1Co&id=4
```

### Get Accounts

#### Command

> * Command: `sdk_sign`

Reviewer: Should we put "Get Accounts" in front of "Sign"?
```suggestion
* Command: `sdk_get_accounts`
```

Author: Good catch

* Parameters:
  * `coins`: BIP44 code array encoded as URL params. Check how it's encoded in [Dictionary Encoding](#dictionary-encoding).

##### Example
```
trust://sdk_get_accounts?coins.0=60&coins.1=714&app=sampleapp&callback=sdk_sign_result&id=2
```

#### Response
* Parameters:
  * `accounts`: Public addresses joined by `,`.

##### Example
```

> sampleapp://sdk_sign_result?accounts=0xFC6CD054fAc48Df3744B42ea82E1Dd0fa7027086,

Reviewer: typo sdk_sign_result ?
